controllers/grafanacontactpoint_controller.go

```diff
@@ -0,0 +1,286 @@
+		}
+	}
+	condition := metav1.Condition{
+		Type:               conditionContactPointSynchronized,
+		ObservedGeneration: contactPoint.Generation,
+		LastTransitionTime: metav1.Time{
+			Time: time.Now(),
+		},
+	}
+
+	if len(applyErrors) == 0 {
+		condition.Status = "True"
+		condition.Reason = "ApplySuccesfull"
+		condition.Message = fmt.Sprintf("Contact point was successfully applied to %d instances", len(instances))
+	} else {
+		condition.Status = "False"
+		condition.Reason = "ApplyFailed"
+
+		var sb strings.Builder
+		for i, err := range applyErrors {
+			sb.WriteString(fmt.Sprintf("\n- %s: %s", i, err))
+		}
+
+		condition.Message = fmt.Sprintf("Contact point failed to be applied for %d out of %d instances. Errors:%s", len(applyErrors), len(instances), sb.String())
+	}
+	meta.SetStatusCondition(&contactPoint.Status.Conditions, condition)
+
+	return ctrl.Result{RequeueAfter: contactPoint.Spec.ResyncPeriod.Duration}, nil
+}
+
+func (r *GrafanaContactPointReconciler) reconcileWithInstance(ctx context.Context, instance *grafanav1beta1.Grafana, contactPoint *grafanav1beta1.GrafanaContactPoint) error {
+	cl, err := client2.NewGeneratedGrafanaClient(ctx, r.Client, instance)
+	if err != nil {
+		return fmt.Errorf("building grafana client: %w", err)
+	}
+
+	applied, err := r.getContactPointFromUID(ctx, instance, contactPoint)
+	if err != nil {
+		return fmt.Errorf("getting contact point by UID: %w", err)
+	}
+
+	if applied.UID == "" {
+		// create
+		cp := &models.EmbeddedContactPoint{
+			DisableResolveMessage: contactPoint.Spec.DisableResolveMessage,
+			Name:                  contactPoint.Spec.Name,
+			Type:                  &contactPoint.Spec.Type,
+			Settings:              contactPoint.Spec.Settings,
+		}
+		result, err := cl.Provisioning.PostContactpoints(provisioning.NewPostContactpointsParams().WithBody(cp))
+		if err != nil {
+			return fmt.Errorf("creating contact point: %w", err)
+		}
+
+		contactPoint.Status.ContactPointUID = result.Payload.UID
+	}
+	if applied.UID != "" {
+		// update
+		var updatedCP models.EmbeddedContactPoint
+		updatedCP.Name = contactPoint.Spec.Name
+		updatedCP.Type = &contactPoint.Spec.Type
+		updatedCP.Settings = contactPoint.Spec.Settings
+		_, err := cl.Provisioning.PutContactpoint(provisioning.NewPutContactpointParams().WithUID(applied.UID).WithBody(&updatedCP)) //nolint:errcheck
+		if err != nil {
+			return fmt.Errorf("updating contact point: %w", err)
+		}
+	}
+	return nil
+}
+
+func (r *GrafanaContactPointReconciler) getContactPointFromUID(ctx context.Context, instance *grafanav1beta1.Grafana, contactPoint *grafanav1beta1.GrafanaContactPoint) (models.EmbeddedContactPoint, error) {
+	cl, err := client2.NewGeneratedGrafanaClient(ctx, r.Client, instance)
+	if err != nil {
+		return models.EmbeddedContactPoint{}, fmt.Errorf("building grafana client: %w", err)
+	}
+
+	params := provisioning.NewGetContactpointsParams().WithName(&contactPoint.Spec.Name)
```

Comment:
Is there a reason to restrict the contact point query to the current name? This causes renaming of contact points to create a new contact point instead